### ISSUE TITLE
[iOS] Animation stretches out from the top left AnimationView corner since Lottie 4.3.0

### ISSUE THREAD

**Issue by aureliencolas** [airbnb/lottie-ios#2193]
## Issue
Since Lottie 4.3.0, when the animation starts, it is most of the time stretched from the top left corner of its AnimationView to the entier AnimationView (according to the AnimationView contentMode).
The issue described here is a regression introduced with Lottie 4.3.0 as it is not reproduced with Lottie 4.2.0 and earlier.

## Which Version of Lottie are you using?
Lottie 4.3.0 iOS SDK from SPM.

## Expected Behavior 
The animation fills the AnimationView right at the beginning of the animation. No additional stretching animation occurs.

With Lottie 4.2.0, this is the case:
https://github.com/airbnb/lottie-ios/assets/82930766/4dd31b8a-006e-4c1b-a8c1-e63b7cd86cd9

## Actual Behavior
Since Lottie 4.3.0, the animation sometimes starts by streching from the top left corner of its AnimationView to the entire AnimationView.

Here is the issue with Lottie 4.3.0:
https://github.com/airbnb/lottie-ios/assets/82930766/e969896c-97cc-4e98-9fa7-12e341e17249

## Animation JSON
I manage multiple apps and the issue appears with all of them. Here is one of these JSON:
[data-ipad_110.json.txt](https://github.com/airbnb/lottie-ios/files/12642930/data-ipad_110.json.txt)

## Implementation details
In case it is relevant, here is how the AnimationView is added to a UIViewController:
```swift
private let animationView = LottieAnimationView()

override func viewDidLoad() {
  super.viewDidLoad()

  animationView.translatesAutoresizingMaskIntoConstraints = false
  view.addSubview(animationView)
  animationView.constrain(to: view) // helper to constrain the for view edges to another view
  animationView.backgroundBehavior = .forceFinish
  animationView.backgroundColor = UIColor(named: "content_background1")
  animationView.contentMode = .scaleAspectFit
  animationView.isHidden = true
}
```
Then, when the animation has to start, here is how it is started:
```swift
animationView.isHidden = false
animationView.animation = animation
animationView.play(completion: completion)
```

I tried to initialise the AnimationView with UIView.performWithAnimation() but with no luck. Actually, the AnimationView itself is already constrain when it appears; it is the animation itself which stretches out, not the AnimationView.

**Comment by calda** [comment]
Can you share a zip of sample Xcode project that demonstrates this issue? Thanks!

**Comment by aureliencolas** [comment]
I basically stripped the whole project from everything but the main UIViewController and its animation stuff and unfortunately I cannot reproduce the issue anymore... I share it with you anyways:
[LottieIssue2193.zip](https://github.com/airbnb/lottie-ios/files/12643825/LottieIssue2193.zip)

In the real app, since the animation is shown during the app startup which performs heavy background tasks so I wonder if the issue could be related to this heavy workload (the sample app does nothing when the animation is shown).

Besides, I used to set the animation (loaded from the embedded assets with NSDataAsset and JSONDecoder) right before calling `AnimationView.play()` but I figured out that setting the animation much earlier than calling `AnimationView.play()` makes the issue to not occur anymore. Two different ways:
1. Set the animation at `viewDidLoad` and call `AnimationView.play()` after `viewDidAppear`.
2. Still call `AnimationView.play()` right after setting the animation but in an async dispatch.

I don't know if all these information will help you and I am sorry to not provide you with a clear way to reproduce the issue. But it is pretty clear to me that something brought in Lottie 4.3.0 must explain the issue as I definitely don't reproduce it with Lottie 4.2.0.

**Comment by calda** [comment]
Thanks for investigating a bit. I don't think I can investigate further on our end without a sample project that reproduces the issue. If you're able to share one in the future feel free to reopen this issue!

**Comment by calda** [comment]
If you're able to git bisect the commits between 4.2.0 and 4.3.0, and then share which exact commit introduced the issue, that may help point out where the problem is coming from.

**Comment by calda** [comment]
If I had to guess it was probably caused by the commit that refactored `LottieAnimationView` into a `LottieAnimationLayer` CALayer subclass wrapped by a `LottieAnimationView` UIView subclass

**Comment by aureliencolas** [comment]
Dear @calda , here is the commit bringing the regression: dc9316c1094f5095ee84bc71112e47e6e44fc19b (Update text provider API to use full AnimationKeypath values). I hope it helps in any ways.

By the way, I've forked the project in order to bisect the commits so, since I only am able to reproduce the issue, I'm willing to try any fix that comes to your mind.

Thanks.

**Comment by calda** [comment]
If it because I added `display()` to `CoreAnimationLayer.forceDisplayUpdate()`? That change comes to mind as being potentially related

**Comment by calda** [comment]
Great, I'll post a fix.

Are you able to share a before / after gif of the issue so I can attach it to the PR for context?

**Comment by aureliencolas** [comment]
Here is what I came up to:
![Lottie_2193](https://github.com/airbnb/lottie-ios/assets/82930766/26c85e69-f70c-489c-8c05-bbffebfa6a99)

**Comment by calda** [comment]
Releasing this fix in 4.3.1, thanks again for the help!

**Comment by aureliencolas** [comment]
Thanks to you for this amazing framework!
